liblocation/src/main/java/com/mapbox/android/core/location/GoogleLocationEngine.java

```diff
@@ -170,4 +169,22 @@ private void connect() {
   private void updateRequestPriority(LocationRequest request) {
     REQUEST_PRIORITY.get(priority).update(request);
   }
-}
+
+  private void generatePendingIntent(Context context) {
+    pendingIntent = LocationPendingIntentProvider.buildIntent(context, this).retrievePendingIntent();
+  }
+
+  private void setupBroadcastReceiver(Context context) {
+    final String action = "GoogleLocationEngineBroadcast-" + this.toString();
+    IntentFilter filter = new IntentFilter();
+    filter.addAction(action);
+    this.broadcastReceiver = new BroadcastReceiver() {
+      @Override
+      public void onReceive(Context context, Intent intent) {
+        locationIntentHandler.handle(intent, action);
+      }
+    };
+
+    context.registerReceiver(broadcastReceiver, filter);
```

Comment:
This broadcast receiver should be created in the constructor and registered/unregistered whenever location updates are requested/removed.